runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1517,11 +1517,43 @@ MM_ObjectAccessBarrier::copyObjectFields(J9VMThread *vmThread, J9Class *objectCl
 		/* initialize lockword, if present */
 		lockwordAddress = getLockwordAddress(vmThread, destObject);
 		if (NULL != lockwordAddress) {
-			if (J9_ARE_ANY_BITS_SET(J9CLASS_EXTENDED_FLAGS(objectClass), J9ClassReservableLockWordInit)) {
-				J9_STORE_LOCKWORD(vmThread, lockwordAddress, OBJECT_HEADER_LOCK_RESERVED);
+			U_32 lwValue = 0;
+
+			if (vmThread->javaVM->enableGlobalLockReservation) {
+				/* This path is taken when Global Lock Reservation is enabled. */
+				U_32 lockReservationHistory = (U_32)objectClass->lockReservationHistory;
```

Comment:
I believe for `lockReservationHistory` no need to be `U_32`

Reply:
I've now changed that to UDATA.